src/Microsoft.SymbolStore/SymbolStores/HttpSymbolStore.cs

```diff
@@ -216,10 +250,19 @@ public override void Dispose()
         /// <summary>
         /// Returns true if the http status code is temporary or retryable condition.
         /// </summary>
-        protected bool IsRetryableStatus(HttpStatusCode status)
+        protected bool IsRetryableStatus(HttpStatusCode status) => s_retryableStatusCodes.Contains(status);
+
+        private HashSet<SocketError> s_retryableSocketErrors = new HashSet<SocketError>
         {
-            return s_retryableStatusCodes.Contains(status);
-        }
+            SocketError.ConnectionReset,
+            SocketError.ConnectionAborted,
+            SocketError.ConnectionRefused,
```

Comment:
Do we `ConnectionRefused` in the wild? I wonder if we should eventually call `MarkClientFailure();` if we keep hitting that since this the error code that we would get back if the symbol server exists in DNS but doesn't have an HTTP/HTTPS server?

Reply:
I'm still figuring out the exactly list of retryable socket errors.   And yes I'll probably remove ConnectionRefused.

Reply:
Are suggesting we do something IsConnectionResetError and IsConnectionAbortError together? Other than ConnectionRefused and the platform specific checks in those functions, what I have is pretty close.

Would removing ConnectionRefused be enough do you think?